### PR TITLE
Schema corrections in Tabular and Image data schemas

### PR DESCRIPTION
**Pull request by farisdurrani**
# Schema corrections in Tabular and Image data schemas

**What user problem are we solving?**
- Added trainspace id to sqs queue item
- Added data source in Tabular and Image RTK API
- Added train status enum in backends

### CODE REVIEW

**Review by karkir0003 — APPROVED** [review]
conditionally approve on resolving pyright checks

File: backendCore/endpoints/sqs.py [hunk 1]
```diff
@@ -31,27 +31,31 @@ def writeToQueue() -> str:
...
+        uid = request_data["user"]["uid"]
+        create_success = createTrainspaceData(TrainspaceData(trainspace_id, uid))
+
+        if create_success:
+            return send_success(
+                {
+                    "message": "Successfully added your training request to the queue",
+                    "trainspace_id": trainspace_id,
+                }
+            )
```
[the remark this answers is not in the document]
Comment (farisdurrani) [reply backendCore/endpoints/sqs.py:57]: Fixed

File: backendCore/endpoints/sqs.py [hunk 2]
```diff
@@ -31,27 +31,31 @@
...
-            if create_success:
-                return send_success(
-                    {
-                        "message": "Successfully added your training request to the queue",
-                        "trainspace_id": trainspace_id,
-                    }
-                )
-            else:
-                return send_error("Data queued but failed to create trainspace")
+            return send_error("Data queued but failed to create trainspace")
```
[the remark this answers is not in the document]
Comment (farisdurrani) [reply backendCore/endpoints/sqs.py:59]: Fixed

File: backendCore/endpoints/sqs.py [hunk 3]
```diff
@@ -17,7 +18,7 @@
 
 
 @sqs_bp.route("/writeToQueue", methods=["POST"])
-def writeToQueue() -> str:
+def writeToQueue() -> tuple[str, Union(Literal[200], Literal[400])]:
```
[the remark this answers is not in the document]
Comment (farisdurrani) [reply backendCore/endpoints/sqs.py]: Fixed

[the remark this answers is not in the document]
Comment (farisdurrani) [reply backendCore/endpoints/sqs.py]: Fixed

**Pull request merged**